src/tasks/indexing.py

```diff
@@ -34,5 +33,31 @@ def reset_index():
         logger.info(f"Removing collection {collection_name}")
         chroma_client.delete_collection(collection_name)
 
-    # Re-index the data
-    return index_all()
+
+def reset_index_qdrant():
+    from app.engine.vectordbs.qdrant import get_vector_store
+
+    store = get_vector_store()
+    store.client.delete_collection(
+        store.collection_name,
+    )
+    store._create_collection(
+        collection_name=store.collection_name,
+        vector_size=int(os.getenv("EMBEDDING_DIM", 1536)),
+    )
+
+
+def reset_index():
+    vector_store_provider = os.getenv("VECTOR_STORE_PROVIDER", "chroma")
+    if vector_store_provider == "chroma":
+        reset_index_chroma()
+    elif vector_store_provider == "qdrant":
+        reset_index_qdrant()
+    else:
+        raise ValueError(f"Unsupported vector provider: {vector_store_provider}")
```

Comment:
factor out to `reset_vector_store` and make `reset_index_chroma` and `reset_index_qdrant` private inside that new function?

Reply:
I thought that it's able to an IndexManager or VectorStoreManager in the as the controller but not sure we will add more vector stores so i just put the code here. Why do you think they should be private functions?

Reply:
just for capsulation - as a consumer would only call `reset_index` - can of course also be capsulated in a class